frmts/hdf5/bagdataset.cpp

```diff
@@ -1441,6 +1471,708 @@ CPLErr BAGResampledBand::IReadBlock(int nBlockXOff, int nBlockYOff,
+
+    return GDALRasterBand::GetMinimum(pbSuccess);
+}
+
+/************************************************************************/
+/*                             GetMaximum()                             */
+/************************************************************************/
+
+double BAGInterpolatedBand::GetMaximum(int *pbSuccess)
+
+{
+    if (m_bMinMaxSet)
+    {
+        if (pbSuccess)
+            *pbSuccess = TRUE;
+        return m_dfMaximum;
+    }
+
+    return GDALRasterBand::GetMaximum(pbSuccess);
+}
+
+/************************************************************************/
+/*                     BarycentricInterpolation()                       */
+/************************************************************************/
+
+static bool BarycentricInterpolation(double dfX, double dfY,
+                                     const double adfX[3], const double adfY[3],
+                                     double &dfCoord0, double &dfCoord1,
+                                     double &dfCoord2)
+{
+    /* See https://en.wikipedia.org/wiki/Barycentric_coordinate_system */
+    double dfDenom = (adfY[1] - adfY[2]) * (adfX[0] - adfX[2]) +
+                     (adfX[2] - adfX[1]) * (adfY[0] - adfY[2]);
+    if (fabs(dfDenom) < 1e-5)
+    {
+        // Degenerate triangle
+        return false;
+    }
+    double dfMul0X = (adfY[1] - adfY[2]) / dfDenom;
+    double dfMul0Y = (adfX[2] - adfX[1]) / dfDenom;
+    double dfMul1X = (adfY[2] - adfY[0]) / dfDenom;
+    double dfMul1Y = (adfX[0] - adfX[2]) / dfDenom;
+    double dfCstX = adfX[2];
+    double dfCstY = adfY[2];
+    dfCoord0 = dfMul0X * (dfX - dfCstX) + dfMul0Y * (dfY - dfCstY);
+    dfCoord1 = dfMul1X * (dfX - dfCstX) + dfMul1Y * (dfY - dfCstY);
+    dfCoord2 = 1 - dfCoord0 - dfCoord1;
+    return (dfCoord0 >= 0 && dfCoord0 <= 1 && dfCoord1 >= 0 && dfCoord1 <= 1 &&
+            dfCoord2 >= 0 && dfCoord2 <= 1);
+}
+
+/************************************************************************/
+/*                               SQ()                                   */
+/************************************************************************/
+
+static inline double SQ(double v)
+{
+    return v * v;
+}
+
+/************************************************************************/
+/*                             IReadBlock()                             */
+/************************************************************************/
+
+CPLErr BAGInterpolatedBand::IReadBlock(int nBlockXOff, int nBlockYOff,
+                                       void *pImage)
+{
+    HDF5_GLOBAL_LOCK();
+
+    BAGDataset *poGDS = cpl::down_cast<BAGDataset *>(poDS);
+#ifdef DEBUG_VERBOSE
+    CPLDebug(
+        "BAG",
+        "IReadBlock: nRasterXSize=%d, nBlockXOff=%d, nBlockYOff=%d, nBand=%d",
+        nRasterXSize, nBlockXOff, nBlockYOff, nBand);
+#endif
+
+    float *depthsPtr = nullptr;
+    float *uncrtPtr = nullptr;
+
+    GDALRasterBlock *poBlock = nullptr;
+    if (poGDS->nBands == 2)
```

Comment:
Forgive my ignorance, I'm sure there is a good reason. Why handle the case of nBands == 2 specially?

Reply:
That check was actually useless since we always instantiate 2 BAGInterpolatedBand : the first one for depth, the second one for uncertainty. This was a leftover of taking inspiration from BAGResampledBand where there are situations where there is just one single band.
Check replaced by an assertion + comment